src/puppetlabs/puppetdb/http/query.clj

```diff
@@ -304,13 +305,26 @@
                      others parsed))))
       parsed)))
 
+(defn time-and-parse-pql
+  "Time how long it takes to transform a PQL query to AST, if the time
+   is greater than 1s, a warning is logged. Returns the transformed
+   query."
+  [query]
+  (let [start (System/nanoTime)
+        result (pql/parse-pql-query query)
+        elapsed (/ (- (System/nanoTime) start) 1000000.0)
+        max-pql-limit 1000]
+    (if (> elapsed max-pql-limit)
+      (log/warn (trs "Parsing PQL query `{0}` took: `{1}` ms" query elapsed)))
```

Comment:
If you don't feel strongly about it, then suggest removing the backticks for both values, using `pr-str` to formally quote the query, so it'll be easy to unambiguously copy paste from the logs, and putting the query at the end, since it's potentially long, variable length content, and doing so might also make it easier to copy/paste

i.e. I was pondering something like this:
```clj
(log/warn (trs "Parsing PQL took {0} ms: {1}" elapsed (pr-str query)))
```